### ISSUE TITLE
Belos: missing adapter implemetation for Thyra adjoint solve

### ISSUE THREAD

**Issue by mperego** [trilinos/Trilinos#3087]
Belos does not provide an adapter for 
Thyra::solve(A, TRANS, b, x);
see https://github.com/trilinos/Trilinos/blob/master/packages/thyra/core/src/interfaces/operator_solve/fundamental/Thyra_LinearOpWithSolveBase_decl.hpp#L553

It's not urgent because one can obtain the same result by providing the implicitly transposed linear operators A and its preconditioner. But it is a feature that would be nice to have.

@trilinos/belos 

## Your Environment

- **Relevant repo SHA1s:**  
- **Relevant configure flags or configure script:**  
- **Operating system and version:**  
- **Compiler and TPL versions:**  

## Related Issues

* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of

**Comment by mhoemmen** [comment]
Would this involve adding an overload to `Belos::OperatorTraits`?  If so, it's actually nontrivial to do this without breaking backwards compatibility, since OperatorTraits uses compile-time polymorphism.  We do have some experience with that for 64-bit support, though.  @hkthorn would know more.
